com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderUtil.java

```diff
@@ -388,6 +388,68 @@ public static class Result {
         private static final String glFragColorAttrLayoutPrefix = "layout(location = %d) ";
         private static final String floatPrecisionAttrRep       = "precision mediump float;\n";
 
+        private static boolean isOpaqueType(String type) {
+            for( String opaqueTypePrefix : opaqueUniformTypesPrefix) {
+                if(type.startsWith(opaqueTypePrefix)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public static Result transformTextureUniforms(String input) throws CompileExceptionError {
+            Result result = new Result();
+
+            if(input.isEmpty()) {
+                return result;
+            }
+
+            ArrayList<String> output = new ArrayList<>(input.length());
+            ArrayList<String[]> lineReplacements = new ArrayList<>();
+            String[] inputLines = input.split("\\r?\\n");
+
+            for(String line : inputLines) {
```

Comment:
Perhaps a future improvement is to exclude commented out code, as it might cause annoying behaviors if we e.g. potentially cause warnings/errors.
iirc, we've had similar issues with our Lua parsing before.